server/src/main/java/org/elasticsearch/gateway/MetaStateService.java

```diff
@@ -276,28 +265,13 @@ public void cleanupIndex(Index index, long currentGeneration) {
     }
 
     /**
-     * Writes index metadata and updates manifest file accordingly.
-     * Used by tests.
+     * Removes manifest file, global metadata and all index metadata
      */
-    public void writeIndexAndUpdateManifest(String reason, IndexMetaData metaData) throws IOException {
-        long generation = writeIndex(reason, metaData);
-        Manifest manifest = loadManifestOrEmpty();
-        Map<Index, Long> indices = new HashMap<>(manifest.getIndexGenerations());
-        indices.put(metaData.getIndex(), generation);
-        manifest = new Manifest(manifest.getCurrentTerm(), manifest.getClusterStateVersion(), manifest.getGlobalGeneration(), indices);
-        writeManifestAndCleanup(reason, manifest);
-        cleanupIndex(metaData.getIndex(), generation);
-    }
-
-    /**
-     * Writes global metadata and updates manifest file accordingly.
-     * Used by tests.
-     */
-    public void writeGlobalStateAndUpdateManifest(String reason, MetaData metaData) throws IOException {
-        long generation = writeGlobalState(reason, metaData);
-        Manifest manifest = loadManifestOrEmpty();
-        manifest = new Manifest(manifest.getCurrentTerm(), manifest.getClusterStateVersion(), generation, manifest.getIndexGenerations());
-        writeManifestAndCleanup(reason, manifest);
-        cleanupGlobalState(generation);
+    public void deleteAll() throws IOException {
+        MANIFEST_FORMAT.cleanupOldFiles(Long.MAX_VALUE, nodeEnv.nodeDataPaths());
+        META_DATA_FORMAT.cleanupOldFiles(Long.MAX_VALUE, nodeEnv.nodeDataPaths());
+        for (String indexFolderName : nodeEnv.availableIndexFolders()) {
+            INDEX_META_DATA_FORMAT.cleanupOldFiles(Long.MAX_VALUE, nodeEnv.resolveIndexFolder(indexFolderName));
```

Comment:
Should we also delete the now-empty `nodeEnv.resolveIndexFolder(indexFolderName).resolve("_state")`?

Reply:
👍